crates/storage/src/traits/spread.rs

```diff
@@ -14,6 +14,11 @@
 
 use super::KeyPtr;
 
+/// The maximum number of adjunctive storage cells a type may require in
+/// order to be stored in the contract storage with `SpreadLayout`.
+/// The number is arbitrarily chosen. Might need adjustments later.
+pub const MAX_FOOTPRINT: u64 = 16;
```

Comment:
This is not actually true. Also this needs renaming for the same reason. The constant is just used by some types in order to make sure that cleaning up behind them won't become way too expensive. Since we are missing Substrate's storage bulk removal feature we cannot do better than this, yet with our current architecture of `ink_storage`.

A better name would probably be `FOOTPRINT_CLEANUP_THRESHOLD`. And maybe we should set it to 32 because that's kinda Rust's magic number in general.